sdcm/cluster_aws.py

```diff
@@ -253,8 +253,14 @@ def _create_mixed_instances(self, count, interfaces, ec2_user_data, dc_idx, tags
         return instances
 
     def _get_instances(self, dc_idx):
+
+        test_id = self.params.get('test_id', default=None)
+        if not test_id:
+            raise ValueError("test_id should be configured for using reuse_cluster")
+
         ec2 = ec2_client.EC2Client(region_name=self.region_names[dc_idx])
-        return [ec2.get_instance_by_private_ip(ip) for ip in self._node_private_ips]
+        instances = list_instances_aws(tags_dict={'TestId': test_id}, region_name=self.region_names[dc_idx])
```

Comment:
how do you handle different types of nodes (Loader, Monitor)?

Reply:
a very good question...

Reply:
@bentsi no I've added a NodeType and NodeIndex flags, and I'm sorting by the NodeIndex